public/app/plugins/panel/stat/AlignmentFactorLengthEditor.tsx

```diff
@@ -0,0 +1,78 @@
+import React, { useCallback } from 'react';
+import { Select } from '@grafana/ui';
+import { css } from 'emotion';
+import { SelectableValue } from '@grafana/data';
+
+interface Props {
+  value?: string;
+  onChange: (value?: string) => void;
+}
+
+function getLabelText(len: number) {
+  if (len === 0) {
+    return 'empty';
+  }
+  return `${len} - ` + new Array(len + 1).join('x');
+}
+
+const lengths: Array<SelectableValue<number>> = Array(51)
+  .fill(0)
+  .map((_, i) => {
+    return {
+      value: i,
+      label: getLabelText(i),
+    };
+  });
+
+export const AlignmentFactorLengthEditor: React.FC<Props> = ({ value, onChange }) => {
+  const styles = {
+    textInput: css`
+      margin-bottom: 8px;
+    `,
+  };
```

Comment:
nit: maybe move styles function outside of the component and use `useStyles` hook